lib/supplejack_common/paginated_collection.rb

```diff
@@ -95,11 +101,18 @@ def increment_page_counter!
     end
 
     def more_results?
+      if @tokenised
+        return self.klass.next_page_token(@next_page_token_location).present?
+      end
       current_page <= total_pages
     end
 
     def paginated?
-      page && per_page
+      (page && per_page) || @tokenised
+    end
+
+    def tokenised?
```

Comment:
Can @tokenised only be true or false?

Reply:
YES

Reply:
Ah true, yeah it's not clear what the value of @tokenised could be.